### PR TITLE
Add contributing information.

### PR DESCRIPTION
**Pull request by tammoippen**
A link to this file will show up whenever a new Pull Request on Github is created. Also see [Contributing Guidelines](https://github.com/blog/1184-contributing-guidelines) for this topic. We could also think about a PR template as suggested [here](https://github.com/blog/2111-issue-and-pull-request-templates).

Most of the content is copied from the [developer space](https://nest.github.io/nest-simulator/) and some structure is inspired from [puppet](https://github.com/puppetlabs/puppet/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+* In general, the rule is that each pull request needs an OK from the CI platform and at least two reviewers to be merged.
+* For changes labeled “not code” or “minor” (e.g. changes in documentation, fixes for typos, etc.), the release manager can waive the need for code review and just accept the OK from Travis in order to merge the request.
+* Each pull request needs to be documented by an issue in the [issue tracker](https://github.com/nest/nest-simulator/issues) explaining the reason for the changes and the solution. The issue is also the place for discussions about the code.
+* New features like SLI or PyNEST functions, neuron or synapse models need to be accompanied by one or more tests written either in SLI or Python. New features for the NEST kernel need a test written in SLI.
+* Each change to the code has to be reflected also in the corresponding examples and documentation.
+* All source code has to be adhering to the Coding Guidelines for [C++](https://nest.github.io/nest-simulator/coding_guidelines_c++) and [SLI](https://nest.github.io/nest-simulator/coding_guidelines_sli) in order to pass the continuous integration system checks.
```
Comment (abigailm) [inline CONTRIBUTING.md]: coding guidelines

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+* All Commits should be coherent and contain only changes that belong together.
```
Comment (abigailm) [inline CONTRIBUTING.md]: commits

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Submitting Changes
+
+* Sign the [Contributor License Agreement](https://nest.github.io/nest-simulator/#contributor-license-agreement).
+* Push your changes to a topic branch in your fork of the repository.
+* Submit a pull request to the [NEST repository](https://github.com/nest/nest-simulator).
+* The core team looks at Pull Requests on a regular basis and posts feedback.
```
Comment (abigailm) [inline CONTRIBUTING.md]: pull requests

**Comment by abigailm** [comment]
Great except for some Excessive Capitalisation

**Comment by abigailm** [comment]
:+1:  from me but maybe @heplesser has some additional feedback

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
+# How to contribute
+
+The NEST simulator is a scientific tool and as such it is never ready and constantly changing to meet the needs of novel neuroscientific endeavors. Here you find the the most important information on how you can contribute to NEST. This document is an excerpt from our [developer space](https://nest.github.io/nest-simulator/), which provides more detailed information.
```
Comment (jougs) [inline CONTRIBUTING.md]: double "the" in "find the the most important"

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+## Getting started
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* The development workflow is based purely on pull requests. This [article](https://nest.github.io/nest-simulator/development_workflow) gives information on how to work with git and GitHub, if you are new to it.
+
+## Making Changes
+
+* Create a topic branch from NEST master in your fork. Please avoid working directly on the `master` branch.
```
Comment (jougs) [inline CONTRIBUTING.md]: Add a reference to https://github.com/nest/nest-simulator/issues/31 to explain why. Or even better integrate that information into the developer space!

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+* Make sure all tests pass (`make installcheck`).
+* Make sure your code conforms to our [coding guidelines](https://nest.github.io/nest-simulator/coding_guidelines_c++#check-static-analysis-local) (`./extras/check_code_style.sh`)
+
+
+## Code Review
+
+We review each pull request according to our [code review guidelines](https://nest.github.io/nest-simulator/code_review_guidelines):
+
+* In general, the rule is that each pull request needs an OK from the CI platform and at least two reviewers to be merged.
+* For changes labeled “not code” or “minor” (e.g. changes in documentation, fixes for typos, etc.), the release manager can waive the need for code review and just accept the OK from Travis in order to merge the request.
```
Comment (jougs) [inline CONTRIBUTING.md]: Shouldn't these be normal quotes (around "not code" and "minor"? They look like some kind of fancy unicode quotes to me that might cause trouble on some systems

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+* Each pull request needs to be documented by an issue in the [issue tracker](https://github.com/nest/nest-simulator/issues) explaining the reason for the changes and the solution. The issue is also the place for discussions about the code.
```
Comment (jougs) [inline CONTRIBUTING.md]: I think we can remove this sentence both from here and from the developer space. Almost no one is adhering to this and I personally have the feeling that this causes more work and separation of information than it helps to keep the discussion structured (which was the original aim of this requirement).

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+* New features like SLI or PyNEST functions, neuron or synapse models need to be accompanied by one or more tests written either in SLI or Python. New features for the NEST kernel need a test written in SLI.
+* Each change to the code has to be reflected also in the corresponding examples and documentation.
+* All source code has to be adhering to the coding guidelines for [C++](https://nest.github.io/nest-simulator/coding_guidelines_c++) and [SLI](https://nest.github.io/nest-simulator/coding_guidelines_sli) in order to pass the continuous integration system checks.
+* All commits should be coherent and contain only changes that belong together.
```
Comment (jougs) [inline CONTRIBUTING.md]: Please add a sentence about the adherence to PEP-8 for Python files

**Comment by jougs** [comment]
Please consider waiting for answers to my comments before merging.

File: CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,48 @@
+# How to contribute
+
+The NEST simulator is a scientific tool and as such it is never ready and constantly changing to meet the needs of novel neuroscientific endeavors. Here you find the the most important information on how you can contribute to NEST. This document is an excerpt from our [developer space](https://nest.github.io/nest-simulator/), which provides more detailed information.
+
+## Getting started
+
+* Make sure you have a [GitHub account](https://github.com/signup/free)
+* The development workflow is based purely on pull requests. This [article](https://nest.github.io/nest-simulator/development_workflow) gives information on how to work with git and GitHub, if you are new to it.
```
Comment (heplesser) [inline CONTRIBUTING.md]: No comma before "if you are new to it".

File: CONTRIBUTING.md (hunk 7 above)
Comment (heplesser) [inline CONTRIBUTING.md]: I agree!

File: CONTRIBUTING.md [hunk 10]
```diff
@@ -0,0 +1,48 @@
...
+
+* In general, the rule is that each pull request needs an OK from the CI platform and at least two reviewers to be merged.
+* For changes labeled “not code” or “minor” (e.g. changes in documentation, fixes for typos, etc.), the release manager can waive the need for code review and just accept the OK from Travis in order to merge the request.
+* Each pull request needs to be documented by an issue in the [issue tracker](https://github.com/nest/nest-simulator/issues) explaining the reason for the changes and the solution. The issue is also the place for discussions about the code.
+* New features like SLI or PyNEST functions, neuron or synapse models need to be accompanied by one or more tests written either in SLI or Python. New features for the NEST kernel need a test written in SLI.
+* Each change to the code has to be reflected also in the corresponding examples and documentation.
+* All source code has to be adhering to the coding guidelines for [C++](https://nest.github.io/nest-simulator/coding_guidelines_c++) and [SLI](https://nest.github.io/nest-simulator/coding_guidelines_sli) in order to pass the continuous integration system checks.
```
Comment (heplesser) [inline CONTRIBUTING.md]: "has to conform" instead of "has to be adhering"

File: CONTRIBUTING.md [hunk 11]
```diff
@@ -0,0 +1,47 @@
...
+* Make sure NEST compiles and has no new warnings.
+* Make sure all tests pass (`make installcheck`).
+* Make sure your code conforms to our [coding guidelines](https://nest.github.io/nest-simulator/coding_guidelines_c++#check-static-analysis-local) (`./extras/check_code_style.sh`)
+
+
+## Code Review
+
+We review each pull request according to our [code review guidelines](https://nest.github.io/nest-simulator/code_review_guidelines):
+
+* In general, the rule is that each pull request needs an OK from the CI platform and at least two reviewers to be merged.
```
Comment (abigailm) [inline CONTRIBUTING.md]: perhaps it's worth mentioning that *if* it is relevant to an issue then it should say so explicitly (now we have removed the line about requiring an issue number)

**Comment by tammoippen** [comment]
@abigailm Please see 795134b.

**Comment by jougs** [comment]
I think that all comments have been sufficiently addressed. Many thanks for providing this. :+1: and merging.

**Pull request merged**